sdk/trace/tracer.go

```diff
@@ -39,17 +39,15 @@ func (tr *tracer) Start(ctx context.Context, name string, o ...apitrace.SpanOpti
 		op(&opts)
 	}
 
-	// TODO: [rghetia] ChildOfRelationship is used to indicate that the parent is remote
-	// and its context is received as part of a request. There are two possibilities
-	// 1. Remote is trusted. So continue using same trace.
-	//      tracer.Start(ctx, "some name", ChildOf(remote_span_context))
-	// 2. Remote is not trusted. In this case create a root span and then add the remote as link
-	//      span := tracer.Start(ctx, "some name")
-	//      span.Link(remote_span_context, ChildOfRelationship)
-	if opts.Reference.SpanContext != core.EmptySpanContext() &&
-		opts.Reference.RelationshipType == apitrace.ChildOfRelationship {
-		parent = opts.Reference.SpanContext
-		remoteParent = true
+	if reference := opts.Reference; reference.SpanContext != core.EmptySpanContext() {
```

Comment:
Reference is not defined in the specs. What does a follow from mean?

Reply:
Then we should remove `FollowsFrom` completely.

Reply:
Reference should be replaced simply with SpanContext in [SpanOptions](https://github.com/open-telemetry/opentelemetry-go/blob/3516ebc456c7ce437d6c12071e44d90b029fc2aa/api/trace/api.go#L102-L106) and add 'remoteParent bool' in [SpanContext](https://github.com/open-telemetry/opentelemetry-go/blob/3516ebc456c7ce437d6c12071e44d90b029fc2aa/api/core/span_context.go#L36-L40). Propagators should set it to true.
With remoteParent bool in SpanContext, there is no need for '*remote*' parameters/fields in bunch of places.

Reply:
Why should `remoteParent` go in the span context? If you start a span with an explicit `ChildOf` or `FollowsFrom`, isn't that a sufficient indicator? I.e., this bit can be computed and stored in the SDK if needed.

Reply:
Folks, remember that we have a OpenTracing compatibility story. OpenTracing defined FollowsFrom.

Reply:
I agree with @jmacd on both counts, particularly about not adding `remoteParent` to `SpanContext` - that feels to me like it would introduce more room for confusion. If we want to make the "remote" vs "local" aspect more explicit, what about something like renaming the start options so that they clearly indicate they're meant for remote parents only (e.g., `RemoteChildOf`)?

I also feel like this conversation should be moved outside of this PR, which is meant to be about what seems like a relatively contained bug based on the existing code.

Reply:
I thought we were using the word "Relation", not "Reference", but that's not something for this PR. I see that StartSpanOptions has a Reference, but I think it would be clearer if we used Relation or Relationship.

Reply:
> Why should `remoteParent` go in the span context? If you start a span with an explicit `ChildOf` or `FollowsFrom`, isn't that a sufficient indicator? I.e., this bit can be computed and stored in the SDK if needed.

[Spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#spancontext) makes it seem like 'remoteParent' is part of SpanContext. 

Anyway, as far as this PR goes the change looks fine.